app/assets/javascripts/discourse/app/lib/media-optimization-utils.js

```diff
@@ -17,9 +17,16 @@ function drawableToImageData(drawable) {
   if (!ctx) {
     throw "Could not create canvas context";
   }
+
+  if (width * height > 67108864) {
```

Comment:
This limit will apply to all browsers. I hesitate to pass capabilities' data to this helper, in regular usage, this shouldn't be hit very often.